src/NuGet.Clients/PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -74,12 +80,12 @@ private void OnSolutionOpenedOrClosed(object sender, EventArgs e)
 
         public bool DeleteSection(string section)
         {
-            return SolutionSettings.DeleteSection(section);
+            return CanChangeSettings && SolutionSettings.DeleteSection(section);
```

Comment:
Can we show a warning message or error message when actions like delete, set, update are not working ?